conans/client/tools/win.py

```diff
@@ -237,7 +309,17 @@ def find_windows_10_sdk():
 def vcvars_command(settings, arch=None, compiler_version=None, force=False, vcvars_ver=None,
                    winsdk_version=None):
     arch_setting = arch or settings.get_safe("arch")
-    compiler_version = compiler_version or settings.get_safe("compiler.version")
+
+    compiler = settings.get_safe("compiler")
+    if compiler == 'Visual Studio':
+        compiler_version = compiler_version or settings.get_safe("compiler.version")
+    else:
+        # vcvars might be still needed for other compilers, e.g. clang-cl or Intel C++,
+        # as they might be using Microsoft STL and other tools (e.g. resource compiler, manifest tool, etc)
+        # in this case, use the latest Visual Studio available on the machine
+        _, last_version = latest_visual_studio_version_installed(output=_global_output)
```

Comment:
`latest_visual_studio_version_installed` can return `None`, here we will get:

```
>>> _, _ = None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: cannot unpack non-iterable NoneType object
>>> 
```

Reply:
good catch!